### PR TITLE
feat(RELEASE-981): refactor verify-access-to-resources into a standalone task

### PR DESCRIPTION
**Pull request by seanconroy2021**
feat([RELEASE-981](https://issues.redhat.com//browse/RELEASE-981)): refactor verify-access-to-resources into a standalone task

This PR refactors the `verify-access-to-resources` script from being hard coded in the pipeline to a dedicated Tekton task.

**Changes:**
- Refactored pipelines to utilize the new `verify-access-to-resources` task.
- Removed the inline script from each pipeline and replaced it with a task reference.
- Updated version and documentation to reflect these changes.
- **Task details:**
  - Added a README for the new task.
  - Implemented a test to ensure the task correctly verifies resource accessibility.

**Note:** 
_verifying the no-access scenario test for the task would require creating a new PipelineRun and RoleBindings, followed by updating `test_tekton_tasks.sh`. Although possible, this approach is overly complex. The task has been validated to work as expected because the command `kubectl auth can-i get releaseplanadmission -n <namespace>  --as=system:serviceaccount:default:default` correctly returns "no." Thus, we can trust this verification. Additionally, a test is already in place to check if the resources are accessible. If the task cannot access the resources, the task will exit with `exit 1`._

Signed-off-by: Sean Conroy <sconroy@redhat.com>

### CODE REVIEW

File: application-api
Comment (johnbieren) [inline application-api:1]: this should not be part of the PR

File: pipelines/rh-advisories/rh-advisories.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ kind: Pipeline
 metadata:
   name: rh-advisories
   labels:
-    app.kubernetes.io/version: "0.11.1"
+    app.kubernetes.io/version: "0.12.1"
```
Comment (johnbieren) [inline pipelines/rh-advisories/rh-advisories.yaml]: this should be 0.12.0

File: tasks/verify-access-to-resources/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# verify-access-to-resources
+
+This Tekton task is used to verify access to various resources in the pipelines. It ensures that the necessary resources, such as the release, release plan, release plan admission, release service config and snapshot, are available and accessible. Addiotinaly it check if internal requests can be created.
+
+## Parameters
+
+| Name                 | Description                                        | Optional | Default value |
+|----------------------|----------------------------------------------------|----------|---------------|
+| release              | Namespace/name of the Release                     | No       | -             |
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/README.md]: The spacing is off from here down (needs one more space in the description)

File: tasks/verify-access-to-resources/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+| releasePlan          | Namespace/name of the ReleasePlan                 | No       | -             |
+| releasePlanAdmission | Namespace/name of the ReleasePlanAdmission        | No       | -             |
+| releaseServiceConfig | Namespace/name of the ReleaseServiceConfig        | No       | -             |
+| snapshot             | Namespace/name of the Snapshot                    | No       |
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/README.md]: missing final |

File: tasks/verify-access-to-resources/verify-access-to-resources.yaml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to verify access to required resources & check if internal requests can be created.
+  params:
+    - name: release
+      description:
+        The namespaced name (namespace/name) of the Release custom resource
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/verify-access-to-resources.yaml]: These descriptions should match what you have in the README (I would vote to copy the ones from there to here as the capitalization of the resources is not consistent in the descriptions in this file)

File: tasks/verify-access-to-resources/verify-access-to-resources.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      type: string
+    - name: releasePlan
+      description:
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/verify-access-to-resources.yaml]: If the line doesn't go over (I think it's 120) the character limit, you can have the text on the same line

File: tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+              kubectl apply -f releaseplan
+
+              cat > releaseplanadmission << EOF
+              apiVersion: appstudio.redhat.com/v1alpha1
+              kind: ReleasePlanAdmission
+              metadata:
+                name: releaseplanadmission-sample
+                namespace: default
+              EOF
+              kubectl apply -f releaseplanadmission
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml:46]: I am surprised this works. There are required fields in the spec, but I guess since you omitted the spec struct entirely it is okay. Nice

File: tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+              cat > snapshot << EOF
+              apiVersion: appstudio.redhat.com/v1alpha1
+              kind: Snapshot
+              metadata:
+                name: snapshot-sample
+                namespace: default
+              EOF
+              kubectl apply -f snapshot
+
```
Comment (johnbieren) [inline tasks/verify-access-to-resources/tests/test-verify-access-to-resources-permissions.yaml]: unneeded new line

File: release-service
Comment (johnbieren) [inline release-service:1]: this should not be here

File: internal-services
Comment (johnbieren) [inline internal-services:1]: this should not be part of the pr

**Comment by seanconroy2021** [comment]
Thank you for the review, @johnbieren. I don't know how the internal services, release services, and application API got into the PR.

**Comment by johnbieren** [comment]
Can you squash these all down to one commit? You are failing gitlint

File: pipelines/rh-advisories/rh-advisories.yaml (hunk 1 above)
Comment (johnbieren) [reply pipelines/rh-advisories/rh-advisories.yaml]: You updated the tekton minVersion instead of the pipeline version after my comment

**Review by johnbieren — COMMENTED** [review]
Just the one version thing to fix and squashing the commits then I will approve. One thing to note - typically we let the reviewer resolve their comment once the change is made and they are satisfied, instead of the PR submitter resolving their comments (to avoid situations like the rh-advisories version where the thing wasn't actually fixed)

File: pipelines/rh-advisories/rh-advisories.yaml (hunk 1 above)
Comment (seanconroy2021) [reply pipelines/rh-advisories/rh-advisories.yaml]: Changed it now 👍🏻

**Comment by seanconroy2021** [comment]
> Just the one version thing to fix and squashing the commits then I will approve. One thing to note - typically we let the reviewer resolve their comment once the change is made and they are satisfied, instead of the PR submitter resolving their comments (to avoid situations like the rh-advisories version where the thing wasn't actually fixed)

ooh apologize for resolving the comments myself. I wasn't aware that it was handled that way. I've squashed the commits and fixed the issue with the version. It should be ready to go :)

**Review by johnbieren — APPROVED** [review]
/retest
